### DISCUSSION TITLE
Navigating between full-page components are reloading all the js files

### DISCUSSION THREAD

**Discussion by sayedsadat344** [livewire/livewire#6436] [1. Help]
I have two full-page components in livewire `(Product & Category)` and I am navigating between them using `wire:navigate`.

### Navigate Links
```
 <li><a href="/products" wire:navigate>Product List</a></li>
 <li><a href="/product/category" wire:navigate>Product Categories</a></li>
```

one thing that I have noticed is that, every navigation is` loading all the JavaScript assets`.

See in the screen shots, every time I navigate all the JS files are loaded again.

### Product: 

<img width="523" alt="product" src="https://github.com/livewire/livewire/assets/34087696/26ab76aa-8c34-4f3d-a5d9-48c539fe4fc3">

### Category: 

<img width="527" alt="category" src="https://github.com/livewire/livewire/assets/34087696/0ac4ac2f-3213-4a60-83f2-b018b2da15e3">

My question is that, is this normal behavior or I am not using proper functionality of Livewire V3?

**Comment by PhiloNL** [comment]
That depends where you have placed your scripts:

- Scripts in `<head>` are loaded once
- New `<head>` scripts are evaluated
- Scripts in the `<body>` are re-evaluated (or add `data-navigate-once` to disable re-evaluation)

 See the [docs for more information](https://livewire.laravel.com/docs/navigate#script-evaluation).

**Reply by sayedsadat344** [reply]
It does disables `re-evaluation `but `it stops javascript to function anymore`.

In the same page/component where hot-reload happened everything working fine, but navigating to any component or back to previous component is not having the JavaScript functionality.

What can be the reason?

Note: I did try putting them in `head `section, it wasn't working there too.

**Reply by PhiloNL** [reply]
That's difficult to tell without an example. For me, `data-navigate-once` seems to work, and the javascript that has been evaluated remains available. If you could provide an example repository that replicates the problem or submit a failing test, that would be great.
